Core.IntegrationTests/Git/CommandLineGitClientTests.cs

```diff
@@ -207,6 +207,27 @@ public void GetAncestors_WithTwoExpectedAncestors_ReturnsBothMatchingNames ()
     Assert.That(ancestor, Does.Not.Contain("master"));
   }
 
+  [Test]
+  public void GetAncestors_WithFeatureBranchWithAncestorNameInTitle_DoesNotReturnFeatureBranch()
+  {
+    AddCommit();
+    ExecuteGitCommand("checkout -b develop");
+    AddCommit();
+    ExecuteGitCommand("checkout develop");
+    AddCommit();
+    ExecuteGitCommand("checkout -b feature/something-needs-to-be-developed");
+    AddCommit();
+    ExecuteGitCommand("checkout master");
+
+    var client = new CommandLineGitClient();
+
+    var ancestor = client.GetAncestors("develop");
+
+    Assert.That(ancestor, Does.Contain("develop"));
+    Assert.That(ancestor, Does.Not.Contain("feature/something-needs-to-be-developed"));
+    Assert.That(ancestor, Does.Not.Contain("master"));
```

Comment:
Since you are expecting a known set of values, I recommend using Is.Equivalent(new string[]{...}) or Is.Equal(new string[]{...}). There's also APIs for testing subsets, but we don't need them.

Why? 
1) it requires les test code (we just pass "devleop")
2) We get good diagnostic messages (not sure how Does./Does.Not.Contain print when they fail for collections).
3) we cover the entire assertion in one statement, which means we can't miss somehting and it is more explicit what we expect (no need to mentally aggrate what we expect).